docs/developer/general.md

```diff
@@ -0,0 +1,134 @@
+- `BETA`: denoted with a `b` and a number indicating that it passed the alpha phase of testing and now in second phase of ensure minimal impact on the user. Focuses on performance, and user interactions. 
+- `RELEASE_CANDIDATE`: denote with a `rc` and a number indication it is in the final phase and is ready to be released right away, just needs to be approved on
+
+The version number indicate in the testing phase of potential final release would be: `MAJOR`.`MINOR`.`PATCH`[`DEV`, `ALPHA`, `BETA`, `RELEASE_CANDIDATE`]. Movements between the following testing phases of a release branch can be made by any main developer in pyDARN. However, **final releases are to be approved and preformed by senior developers or pyDARN**
+
+Currently senior developers are: 
+- Marina Schmidt 
+- Ashton Remier 
+- Angeline Burell 
+- Kevin Sterne 
+
+See [Release Guidelines](developer/release_guidelines.md) for more information on the release process of pyDARN. 
+
+## Copyrights and Licensing
+
+pyDARN is licensed under GPL v3.0 this license it not compatible with other licenses so when adding code in pyDARN your code will be licensed under GPL v3.0. 
+As a requirement of the GPL v3.0 license, all authors of their code hold the copyrights to it but are allowing it to be open-sourced in pyDARN. 
+
+When contributing code to pyDARN please add the copyright and author line at the top of the file, for example: 
+```python
+# (C) copyright 2019 SuperDARN Canada, University of Saskatchewan
+# author: Marina Schmidt
+```
+Because pyDARN is collaborative library, shared-copyright is allowed encouraged. If another developer from another institution helped develop the new feature of code then include their copyright line below, for example: 
+```python 
+# (C) copyright 2019 SuperDARN Canada, University of Saskatchewan 
+# author: Marina Schmidt 
+# (C) copyright 2019 SRI International, San Francisco 
+# author: Ashton Remier
+```
+
+In addition to the copyright line, a disclaimer of the license GPL v3.0 needs to be included in every file, for example: 
+
+```python 
+# DISCLAIMER: This code is licensed under GPL v3.0 which can be located in LICENSE.md in the root directory
+```
+
+If you make modifications to the code that constitute you as an author (like bug-fixes, typos, code refactoring, ...) then you will need to make note when the modification was made, by who and a small description, for example: 
+```python
+# Modifications: 
+# 2019-08-20 Marin Schmidt - fixed a bug with range gates presentation in range-time parameter plots 
+```
+
+## Citing and DOI 
+
+In your code please make sure to cite any proprietary software used and articles that your code may be based off of. 
+
+See [Citing pyDARN](user/citing.md) for details on how to cite pyDARN in your own work. 
+
+Data Object Identification (DOI) is a way for pyDARN to be cited in articles. The DOI is created by [zenodo](https://zenodo.org/) for every final release of pyDARN. 
+
+## Deprecations 
```

Comment:
It may be useful to state that a certain period of time (e.g., next minor release) will be used as a warning period before something is removed and a DeprecationWarning will be present during this period of time.